src/core/net/ip6.cpp

```diff
@@ -426,6 +427,11 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
     {
         if (netif != NULL)
         {
+            if (otGetDeviceRole() == kDeviceRoleChild)
```

Comment:
sorry didn't considering this case.
When looking into the packets of one test, it looks a little weird to see SED would forward ping back to the parent which was forwarded from the parent with destination as Realm_Local_All_ThreadNodes.  That's the reason for this update.  Maybe I should come up another proposal or do you think whether it is an issue we should address?